### PR TITLE
govers: migrate to buildGoModule

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Migrating to `buildGoModule` so `buildGoPackage` can be deprecated as cited in #318069

Updated also location of the package to reflect the newest standard

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/go/govers/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule {
+  pname = "govers";
+  version = "unstable-2016-06-23";
```
Comment (katexochen) [inline pkgs/by-name/go/govers/package.nix:9]:
```suggestion
  version = "0-unstable-2016-06-23";
```
conventions for this have also changed. :)

Comment (luftmensch-luftmensch) [reply pkgs/by-name/go/govers/package.nix:9]: Applied changes!

**Comment by katexochen** [comment]
Result of `nixpkgs-review pr 318229` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>57 packages built:</summary>
  <ul>
    <li>airlift</li>
    <li>airlift.dist</li>
    <li>allmark</li>
    <li>apptainer</li>
    <li>apptainer-overriden-nixos</li>
    <li>aws-assume-role</li>
    <li>aws-assume-role.doc</li>
    <li>aws-env</li>
    <li>aws-gate</li>
    <li>aws-gate.dist</li>
    <li>charliecloud</li>
    <li>consul-alerts</li>
    <li>containerpilot</li>
    <li>coyim</li>
    <li>dep2nix</li>
    <li>devcontainer</li>
    <li>docker</li>
    <li>docker-client</li>
    <li>docker-distribution</li>
    <li>docker-gc</li>
    <li>docker-sbom</li>
    <li>docker_25</li>
    <li>docker_26</li>
    <li>docopts</li>
    <li>fn-cli</li>
    <li>glock</li>
    <li>go-symbols</li>
    <li>go-upower-notify</li>
    <li>govers</li>
    <li>grafana_reporter</li>
    <li>guardian-agent</li>
    <li>kontemplate</li>
    <li>librarian-puppet-go</li>
    <li>mailhog</li>
    <li>mgmt</li>
    <li>nex</li>
    <li>notary</li>
    <li>nvidia-docker</li>
    <li>pipework</li>
    <li>prometheus-dovecot-exporter</li>
    <li>prometheus-minio-exporter</li>
    <li>prometheus-surfboard-exporter</li>
    <li>protoc-gen-twirp</li>
    <li>python311Packages.jupyter-repo2docker</li>
    <li>python311Packages.jupyter-repo2docker.dist</li>
    <li>python312Packages.jupyter-repo2docker</li>
    <li>python312Packages.jupyter-repo2docker.dist</li>
    <li>simplehttp2server</li>
    <li>singularity</li>
    <li>singularity-overriden-nixos</li>
    <li>ssm-session-manager-plugin</li>
    <li>syncthing-tray</li>
    <li>tmsu</li>
    <li>udocker</li>
    <li>udocker.dist</li>
    <li>vndr</li>
    <li>xmonad-log</li>
  </ul>
</details>

**Review by msanft — APPROVED** [review]

File: pkgs/by-name/go/govers/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  postPatch = ''
+    go mod init github.com/rogpeppe/govers
+  '';
+
+  src = fetchFromGitHub {
+    owner = "rogpeppe";
+    repo = "govers";
+    rev = "77fd787551fc5e7ae30696e009e334d52d2d3a43";
+    sha256 = "sha256-lpc8wFKAB+A8mBm9q3qNzTM8ktFS1MYdIvZVFP0eiIs=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/govers/package.nix:21]:
```suggestion
  src = fetchFromGitHub {
    owner = "rogpeppe";
    repo = "govers";
    rev = "77fd787551fc5e7ae30696e009e334d52d2d3a43";
    sha256 = "sha256-lpc8wFKAB+A8mBm9q3qNzTM8ktFS1MYdIvZVFP0eiIs=";
  };

  vendorHash = null;

  postPatch = ''
    go mod init github.com/rogpeppe/govers
  '';
```

Comment (luftmensch-luftmensch) [reply pkgs/by-name/go/govers/package.nix:21]: Done!

Comment (luftmensch-luftmensch) [reply pkgs/by-name/go/govers/package.nix]: Done!

File: pkgs/by-name/go/govers/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  postPatch = ''
+    go mod init github.com/rogpeppe/govers
+  '';
+
+  dontRenameImports = true;
+
+  doCheck = false; # fails, silently
+
+  meta = {
+    description = "A tool for rewriting Go import paths";
```
Comment (SuperSandro2000) [inline pkgs/by-name/go/govers/package.nix:29]:
```suggestion
    description = "Tool for rewriting Go import paths";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**